src/server/pfs/server/driver.go

```diff
@@ -1789,17 +1764,41 @@ func (d *driver) listBranch(ctx context.Context, repo *pfs.Repo, reverse bool, c
 	if reverse {
 		opts.Order = col.SortAscend
 	}
-	if repo.Name == "" {
-		if err := d.branches.ReadOnly(ctx).ListRev(branchInfo, opts, listCallback); err != nil {
-			return err
-		}
-	} else {
-		if err := d.branches.ReadOnly(ctx).GetRevByIndex(pfsdb.BranchesRepoIndex, pfsdb.RepoKey(repo), branchInfo, opts, listCallback); err != nil {
+	if err := d.branches.ReadOnly(ctx).ListRev(branchInfo, opts, listCallback); err != nil {
+		return err
+	}
+
+	return sendBis()
+}
+
+func (d *driver) listBranchInTransaction(txnCtx *txncontext.TransactionContext, repo *pfs.Repo, reverse bool, cb func(*pfs.BranchInfo) error) error {
+	// Validate arguments
+	if repo == nil {
+		return errors.New("repo cannot be nil")
+	}
+
+	if err := d.env.AuthServer().CheckRepoIsAuthorizedInTransaction(txnCtx, repo, auth.Permission_REPO_LIST_BRANCH); err != nil {
+		return err
+	}
+
+	// Make sure that the repo exists
+	if repo.Name != "" {
+		if err := d.repos.ReadWrite(txnCtx.SqlTx).Get(pfsdb.RepoKey(repo), &pfs.RepoInfo{}); err != nil {
+			if col.IsErrNotFound(err) {
+				return pfsserver.ErrRepoNotFound{Repo: repo}
+			}
 			return err
 		}
 	}
 
-	return sendBis()
+	opts := &col.Options{Target: col.SortByCreateRevision, Order: col.SortDescend}
+	if reverse {
+		opts.Order = col.SortAscend
+	}
+	branchInfo := &pfs.BranchInfo{}
+	return d.branches.ReadWrite(txnCtx.SqlTx).GetByIndex(pfsdb.BranchesRepoIndex, pfsdb.RepoKey(repo), branchInfo, opts, func(_ string) error {
+		return cb(branchInfo)
```

Comment:
this skips the provenance sorting logic, which does nothing here because branches in a repo can't be provenant on each other